app/controllers/general.php

```diff
@@ -108,14 +99,15 @@ function mapResponseFormatToClass(string $responseFormat): Filter {
     /* 
     * Response format
     */
-    $responseFormatEnvVar = App::getEnv('_APP_SYSTEM_RESPONSE_FORMAT', '');
-    $responseFormatHeader = $request->getHeader('x-appwrite-response-format', '');
-    $responseFormat = empty($responseFormatHeader) ? $responseFormatEnvVar : $responseFormatHeader;
-    if (empty($responseFormat) || ($filter = mapResponseFormatToClass($responseFormat)) == null) {
-        throw new Exception('No filter available for response format : '.$responseFormat, 404);
-    } else {
-        Response::setFilter($filter);   
+    $responseFormat = $request->getHeader('x-appwrite-response-format', App::getEnv('_APP_SYSTEM_RESPONSE_FORMAT', ''));
+    switch($responseFormat) {
+        case version_compare ($responseFormat , '0.6.2', '=<') :
+            Response::setFilter(new V06());
+            break;
+        default:
+            throw new Exception('No filter available for response format : '.$responseFormat, 404);
```

Comment:
Not sure I would throw an exception here, maybe just let the response class use the latest format?

Reply:
But what if we are sent some unrecognised version? Something like 1.3.2 or v1.3.2 . I feel we must inform the user that such a response format doesn't exist.